src/main/java/org/broadinstitute/hellbender/utils/bwa/BwaMemIndexCache.java

```diff
@@ -0,0 +1,51 @@
+package org.broadinstitute.hellbender.utils.bwa;
+
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Manage a global collection of {@link BwaMemIndex} instances.
+ */
+public class BwaMemIndexCache {
+
+    private final static Map<String, BwaMemIndex> instances = new HashMap<>();
+
+    /**
+     * Returns a {@link BwaMemIndex} instance that corresponds to  given index image file.
+     * @param indexImageFile the target image file.
+     * @return never {@code null}.
+     */
+    public static synchronized BwaMemIndex getInstance( final String indexImageFile ) {
+        Utils.nonNull(indexImageFile, "the index image file name provided cannot be null");
+        if (!instances.containsKey(indexImageFile)) {
+            instances.put(indexImageFile, new BwaMemIndex(indexImageFile));
+        }
+        return instances.get(indexImageFile);
+    }
+
+    /**
+     * Closes all instances in the VM.
+     */
+    public static synchronized void closeInstances() {
+        instances.values().forEach(BwaMemIndex::close);
+        instances.clear();
```

Comment:
It strikes me as a little dangerous to close each one, and only then to clear the Map.  I think a more recoverable way to do this would be to iterate over the map, doing the close and then an iterator.remove().  That way, all successfully closed indexes will have been removed from the map.